### PR TITLE
Write some jsdoc docstrings

### PR DESCRIPTION
**Pull request by IvanSanchez**
I started experimenting with `pc-nrfprog-js`, got annoyed at the lack of documentation, and started writing some jsdoc docstrings.

*Then* I noticed there was a `doc` branch.

This PR uses most of the docstrings from https://github.com/NordicSemiconductor/pc-nrfjprog-js/commit/26620fb81233f6c7e05f094dc7191449ff1c90c2 and https://github.com/NordicSemiconductor/pc-nrfjprog-js/commit/955f08ff6a8360737bac437aaf34d46284c378f5 , plus my best guesses at some stuff and some tiny code examples and formatting.

### CODE REVIEW

File: doc/api.js [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+ *
+ * This is the same functionality as running "<tt>nrfjprog --program</tt>" in the command-line tools.
+ *
+ * @example
+ * nrfjprogjs.program(123456789, "/some/path/nrf52832_abcd.hex", {}, function(err) {
+ *      if (err) throw err;
+ * } );
+ *
+ * @param {integer} serialNumber The serial number of the device to query
+ * @param {string} filename Either the filename of the <tt>.hex</tt> file containing the program, or the contents of such a file.
```
Comment (mrodem) [inline doc/api.js]: Perhaps clarify that the absolute path to the file is required here? Same for the `verify` function.

File: doc/api.js [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+ * Async function to recover a device
+ * <br/>
+ *
+ * Recover the whole chip by removing all user accessible content.
+ * <br/>
+ *
+ * This is the same functionality as running "<tt>nrfjprog --recover</tt>" in the command-line tools.
+ *
+ * @param {integer} serialNumber The serial number of the device to query
+ * @param {module:nrfjprogjs~EraseOptions} filename The filename of the <tt>.hex</tt> file containing the program.
```
Comment (mrodem) [inline doc/api.js]: As far as I can see, there is no `filename` param for recover.

File: doc/api.js [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+ * <br/>
+ *
+ * This is the same functionality as running "<tt>nrfjprog --program</tt>" in the command-line tools.
+ *
+ * @example
+ * nrfjprogjs.program(123456789, "/some/path/nrf52832_abcd.hex", {}, function(err) {
+ *      if (err) throw err;
+ * } );
+ *
+ * @param {integer} serialNumber The serial number of the device to query
```
Comment (IvanSanchez) [inline doc/api.js]: I think I copy-pasted that from the `getDeviceInfo()` docs, where the capabilities are in fact queried, but yeah, the wording is different for each function

File: doc/api.js (hunk 1 above)
Comment (IvanSanchez) [inline doc/api.js]: If https://github.com/NordicSemiconductor/pc-nrfjprog-js/compare/feature/doc#diff-061192fa90b3844e028b074073653c9cR47 doesn't say it has to be absolute, I'm not assuming it has to be absolute

**Pull request merged**